api/types/saml_idp_service_provider.go

```diff
@@ -309,3 +316,14 @@ func (am *SAMLAttributeMapping) CheckAndSetDefaults() error {
 	}
 	return nil
 }
+
+// validatePreset validates SAMLIdPServiceProviderV1 preset field.
+// preset can be either empty or one of the supported type.
+func validatePreset(preset string) bool {
```

Comment:
This PR is not the place to change this, but if you happen to touch this part of the codebase in the future it might be cleaner to move `validatePreset` to the `samlsp` package IMO.